seshat-node/lib/index.js

```diff
@@ -614,6 +614,36 @@ class SeshatRecovery extends seshat.SeshatRecovery {
         return super.info();
     }
 
+    /**
+     * Get the custom user specific version from the database.
+     *
+     * @return {Promise<boolean>} A promise that will resolve to a number that
+     * represents the user version of the database.
+     */
+    async getUserVersion() {
```

Comment:
Huh, I thought this user version stuff was already here...?

Reply:
It is, but it's not exposed through the recovery database which is the only db we can open if the db is marked for a reindex.

Reply:
This might make it clear what's going on https://github.com/vector-im/riot-desktop/pull/104/files#diff-1aa0440c816c324539d613c67b9ea3faR469. 

We can't open the full database because it's marked for a reindex, so we open the recovery database. With this PR we can check the user version from the recovery db as well to check if we can delete the db instead doing the costly reindex.

Reply:
Ah right okay, I just failed to notice this change was for the recovery path. 😅